capsul/study_config/study_config.py

```diff
@@ -306,9 +306,9 @@ def run(self, process_or_pipeline, output_directory= None,
             for name, trait in process_or_pipeline.user_traits().items():
                 if trait.output and isinstance(trait.handler, (File, Directory)):
                     value = getattr(process_or_pipeline, name)
-                    if value is not Undefined and value:
```

Comment:
I fear if it could break with None values. It should not happen but I know we sometimes get None values in addition to Undefined. The test "and value" would filter them out and will no longer, so the next dirname(value) could fail.

Reply:
Right, I restored the check. Thanks!